src/modules/dashboard/components/ColonyHome/ColonyHome.tsx

```diff
@@ -138,66 +135,79 @@ const ColonyHome = ({
     [dispatch, setIsTaskBeingCreated],
   );
 
-  useEffect(() => {
-    const timeout = setTimeout(() => {
-      setRecoverOption(true);
-    }, COLONY_DB_RECOVER_BUTTON_TIMEOUT);
-    return () => clearTimeout(timeout);
-  });
+  /*
+   * @TODO Re-add domains once we decide what we're going to do with the
+   * "Recover Colony" system
+   */
+  // useEffect(() => {
+  //   const timeout = setTimeout(() => {
+  //     setRecoverOption(true);
+  //   }, COLONY_DB_RECOVER_BUTTON_TIMEOUT);
+  //   return () => clearTimeout(timeout);
+  // });
 
   const formSetFilter = useCallback(
     (_: string, value: TasksFilterOptionType) => setFilterOption(value as any),
     [setFilterOption],
   );
 
+  /*
+   * @NOTE Blockchain-first approach
+   * We get the colony's address from the ENS resolver, then using that,
+   * we fetch data from mongo
+   */
   const { data: colonyAddress, error: addressError } = useDataFetcher(
     colonyAddressFetcher,
     [colonyName],
     [colonyName],
   );
 
-  const colonyArgs: [Address | undefined] = [colonyAddress || undefined];
-  const {
-    data: colony,
-    isFetching: isFetchingColony,
-    error: colonyError,
-  } = useDataSubscriber(colonySubscriber, colonyArgs, colonyArgs);
+  const { data: { colony } = {}, loading: colonyDataLoading } = useColonyQuery({
```

Comment:
I am honestly wondering why this even works. The colony address in this query should not be optional (and apparently isn't in the query). You should use a combination of `useEffect` and useColonyLazyQuery (https://github.com/JoinColony/colonyDapp/pull/1932/files#diff-dc754e48e7cb7cefc8a267996180cd40R957) here:

```
  const [loadColony, { data }] = useColonyLazyQuery();

  useEffect(() => {
    if (colonyAddress) {
      loadColony({
        variables: { address: colonyAddress },
      });
    }
  }, [loadColony, colonyAddress]);
```
I know this doesn't look the nicest but it's the safest way for now to ensure the query is using the correct address in cases where it's not defined initially (or can change at some point).

Please also do not use defaults but rather use a loading spinner.